### PR TITLE
Fix a small series of bugs; demo mode, grpid and expected types

### PR DESCRIPTION
**Pull request by jlapenna**

### CODE REVIEW

File: withings_api/__init__.py [hunk 1]
```diff
@@ -318,7 +318,7 @@ def __init__(
         consumer_secret: str,
         callback_uri: str,
         scope: Iterable[AuthScope] = tuple(),
-        mode: str = "demo",
+        mode: str = '',
```
Comment (vangorra) [inline withings_api/__init__.py]: The default should be None.

File: withings_api/common.py [hunk 2]
```diff
@@ -49,6 +49,7 @@ def new_sleep_state(value: Optional[int]) -> SleepState:
 class MeasureGetMeasGroupAttrib(IntEnum):
     """Measure group attributions."""
 
+    UNKNOWN = -1
```
Comment (vangorra) [inline withings_api/common.py]: Interesting. Are you seeing -1 values with data coming back from your account? This value is undocumented in the API spec.

File: withings_api/__init__.py (hunk 1 above)
Comment (jlapenna) [reply withings_api/__init__.py]: Fixed.

File: withings_api/common.py (hunk 2 above)
Comment (jlapenna) [reply withings_api/common.py]: Yeah, Here's a sample: {'grpid': 250284004, 'attrib': -1, 'date': 1409730164  , 'created': 1409730234, 'category': 1, 'deviceid': '0842f39947b59c73cb3662b97dca949b88e2ffa2', 'measures': [{'value': 100, 'type': 9, 'unit': 0, 'algo': 0, 'fw': 0, 'fm': 0}, {'value': 131, 'type': 10, 'unit': 0, 'algo': 0, 'fw': 0,   'fm': 0}, {'value': 94, 'type': 11, 'unit': 0, 'algo': 0, 'fw': 0, 'fm': 0}], 'comment': None}

**Comment by jlapenna** [comment]
I'm new to collaboration on github; let me know if I did this wrong. :)

**Comment by jlapenna** [comment]
Should I fixup the first commit in the PR (3dcae86) with 0a6e0dc; or should I just leave them as is?

**Pull request merged**